File: twoSum.py
class Solution:
    def twoSum(self, nums, target):
        mp = {}
        
        # Map the elements to their indices
        for i, num in enumerate(nums):
            mp[num] = i
        
        for i, num in enumerate(nums):
            complement = target - num
            if complement in mp and mp[complement] != i:
                return [i, mp[complement]]
        
        return []

# Example usage:
solution = Solution()
nums = [2, 7, 11, 15]
target = 9
result = solution.twoSum(nums, target)
if len(result) == 2:
    print("Indices:", result[0], result[1])
else:
    print("No solution found.")

File: twoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
         unordered_map<int,int>mp;

       //map the elements to their indices
         for(int i=0;i<nums.size();i++){
               mp[nums[i]]=i;
         }

       for(int i=0;i<nums.size();i++){
              if(mp[target - nums[i]] and (mp[target - nums[i]] != i)){
                   return {i,mp[target-nums[i]]};
              }
       }

       return {};
    }
};